pkg/client/auditor/auditor.go

```diff
@@ -380,6 +383,37 @@ func (a *defaultAuditor) audit() error {
 	return noErr
 }
 
+func (a *defaultAuditor) verifyStateSignature(
+	serverID string,
+	serverState *schema.ImmutableState,
+) error {
+
+	if a.serverSigningPubKey != nil && serverState.GetSignature() == nil {
+		return fmt.Errorf(
+			"a server signing public key has been specified for the auditor, "+
+				"but the state %s at TX %d received from server %s @ %s is not signed",
+			serverState.GetTxHash(), serverState.GetTxId(), serverID, a.serverAddress)
+	}
+
+	if serverState.GetSignature() != nil {
+		pk := a.serverSigningPubKey
+		if pk == nil {
+			x, y := elliptic.Unmarshal(elliptic.P256(), serverState.GetSignature().GetPublicKey())
```

Comment:
Can we issue some kind of warning here? Checking the signature here without any form of public key verification does not protect from man-in-the-middle attacks and may give a false sense of security.

Reply:
@padurean We'll be merging the PR as is, please prepare another PR that fixes this and @mmeloni 's comments